mllib/src/main/scala/org/apache/spark/mllib/evaluation/MulticlassMetrics.scala

```diff
@@ -129,86 +135,199 @@ class MulticlassMetrics @Since("1.1.0") (predictionAndLabels: RDD[(Double, Doubl
   }.sum
 
   /**
-   * Returns weighted averaged precision
-   */
+    * Returns weighted averaged precision
+    */
   @Since("1.1.0")
   lazy val weightedPrecision: Double = labelCountByClass.map { case (category, count) =>
     precision(category) * count.toDouble / labelCount
   }.sum
 
   /**
-   * Returns weighted averaged f-measure
-   * @param beta the beta parameter.
-   */
+    * Returns weighted averaged f-measure
+    *
+    * @param beta the beta parameter.
+    */
   @Since("1.1.0")
   def weightedFMeasure(beta: Double): Double = labelCountByClass.map { case (category, count) =>
     fMeasure(category, beta) * count.toDouble / labelCount
   }.sum
 
   /**
-   * Returns weighted averaged f1-measure
-   */
+    * Returns weighted averaged f1-measure
+    */
   @Since("1.1.0")
   lazy val weightedFMeasure: Double = labelCountByClass.map { case (category, count) =>
     fMeasure(category, 1.0) * count.toDouble / labelCount
   }.sum
 
   /**
-   * Returns the sequence of labels in ascending order
-   */
+    * Returns the sequence of labels in ascending order
+    */
   @Since("1.1.0")
   lazy val labels: Array[Double] = tpByClass.keys.toArray.sorted
+
+
+  /**
+    * Returns unweighted Cohen's Kappa
+    * Cohen's kappa coefficient is a statistic which measures inter-rater
+    * agreement for qualitative (categorical) items. It is generally thought
+    * to be a more robust measure than simple percent agreement calculation,
+    * since kappa takes into account the agreement occurring by chance.
+    * The kappa score is a number between -1 and 1. Scores above 0.8 are
+    * generally considered good agreement; zero or lower means no agreement
+    * (practically random labels).
+    */
+  @Since("1.6.0")
+  def kappa(): Double = {
+    kappa("default")
+  }
+
+  /**
+    * Returns Cohen's Kappa with built-in weighted types
+    *
+    * @param weights the weighted type. "default" means no weighted;
+    *                "linear" means linear weighted;
+    *                "quadratic" means quadratic weighted.
+    */
+  @Since("1.6.0")
+  def kappa(weights: String): Double = {
+
+    val func = weights match {
+      case "default" =>
+        (i: Int, j: Int) => {
+          if (i == j) {
+            0.0
+          } else {
+            1.0
+          }
+        }
+      case "linear" =>
+        (i: Int, j: Int) => Math.abs(i - j).toDouble
+      case "quadratic" =>
+        (i: Int, j: Int) => (i - j).toDouble * (i - j)
+      case t =>
+        throw new IllegalArgumentException(
+          s"kappa only supports {linear, quadratic, default} but got type ${t}.")
+    }
+
+    kappa(func)
+  }
+
+
+  /**
+    * Returns Cohen's Kappa with user-defined weight matrix
+    *
+    * @param weights the weight matrix, must be of the same shape with Confusion Matrix.
+    *                Note: Each Element in it must be no less than zero.
+    */
+  @Since("1.6.0")
+  def kappa(weights: Matrix): Double = {
```

Comment:
I trend to keep this api, for the input matrix is small, whose size is nClass * nClass.
And in the ﻿relative statistical documents (such as http://www.real-statistics.com/reliability/weighted-cohens-kappa/), there exists the weighted matrix. Keeping it may make kappa score more comprehensive.